Sources/SwiftPhoenixClient/PhoenixTransport.swift

```diff
@@ -202,6 +207,23 @@ open class URLSessionTransport: NSObject, PhoenixTransport, URLSessionWebSocketD
     self.task?.resume()
   }
   
+  open func connect(with headers: [String : Any]) {
+    // Set the transport state as connecting
+    self.readyState = .connecting
+    
+    // Create the session, request, and websocket task
+    self.session = URLSession(configuration: self.configuration, delegate: self, delegateQueue: nil)
+    var request = URLRequest(url: url)
+    for header in headers {
+        guard let value = header.value as? String else { return }
+        request.addValue(value, forHTTPHeaderField: header.key)
+    }
+    self.task = self.session?.webSocketTask(with: request)
+    
+    // Start the task
+    self.task?.resume()
```

Comment:
This and the change below is a lot of duplicate code. Here, `connect()` should call `connect(with: [:])` which will loop over an empty map, add no additional headers and continue on.

As for below, I could see changing the signature to have default empty headers so that the method signature doesn't change for existing users but also we avoid adding a duplicate function

Reply:
I didn't combine them initially because this function is almost guaranteed to be called by anyone using the framework and I wanted to avoid tampering with the original one for that reason. That said, if you're comfortable with it I'm happy to consolidate the two functions!

Reply:
I think that would be ideal. The change should be transparent to users when they update their client due to the default parameters

Reply:
@dsrees updated as discussed.